pkg/pgmodel/querier/query_sample.go

```diff
@@ -124,8 +138,14 @@ func fetchMultipleMetricsSamples(ctx context.Context, tools *queryTools, metadat
 
 	// TODO this assume on average on row per-metric. Is this right?
 	results := make([]sampleRow, 0, len(metrics))
+
+	if len(metrics) == 0 {
+		return results, nil
+	}
```

Comment:
If metrics is empty there's no reason to continue with the code. Before adding the timeout, it created an empty batch and tried to send it. Without this early return, if metrics is empty it will create a batch with just statement_timeout, so we return early and save the request.